### PR TITLE
Provide proper handling of arbitrary length integers

### PR DESCRIPTION
**Pull request by jkramarz**
Current version of library silently fails on numbers bigger than 32^2-1, but does require libraries which are needed only in this case.

Results of encode() and decode() in this pull request are consistent with [davidaurelio/hashids-python](https://github.com/davidaurelio/hashids-python). encodeHex() and decodeHex() is still missing, as dechex/hexdec needs to be reimplemented.

### CODE REVIEW

File: src/Math.php [hunk 1]
```diff
@@ -19,53 +19,121 @@
...
+        return bcadd($a, $b);
     }
 
     /**
-     * Divide two arbitrary precision numbers.
+     * Multiply two arbitrary-length integers
      *
-     * @param int $a
-     * @param int $b
+     * @param string $base
```
Comment (vinkla) [inline src/Math.php]: This should say `$a`.

**Comment by vinkla** [comment]
Thanks for this pull request @jkramarz! Please see my inline comments. Also, we need to fix the code coverage since it dropped with 38%.

**Comment by jkramarz** [comment]
OK, I'll try to configure Travis to run both, BC and GMP, versions first.

**Comment by jkramarz** [comment]
No way to do so - GMP is compiled-in. What do you think, @vinkla, about implementing a way to switch math library implementations? Besides simplifying tests, if one doesn't need bigint support, native operations seems to be much faster. If it is OK for you, which way of switching libraries would you prefer?

**Comment by vinkla** [comment]
@jkramarz maybe we could enable/disable the extension like this? https://docs.travis-ci.com/user/languages/php#Enabling-preinstalled-PHP-extensions

**Comment by jkramarz** [comment]
Already tried, it's not pre-installed but compiled-in.

**Comment by vinkla** [comment]
Oh, too bad. Add what you suggested in earlier comment and then we can discuss it further.

**Comment by jkramarz** [comment]
OK, #76 will do the job for now.

**Comment by jkramarz** [comment]
OK, encodeHex() and decodeHex() uses 12-character chunks, so it fits in 64-bit signed integer, so there's no need to add custom dechex/hexdec implementation. On 32-bit platform original library doesn't work in original version either, so it's another story.

**Comment by vinkla** [comment]
All looks good, lets just merge #76 first and then we can rebase this one to get better coverage.

**Comment by vinkla** [comment]
Awesome work @jkramarz! Thanks for fixing it.

**Comment by ivanakimov** [comment]
This is great 👍

**Pull request merged**